### PR TITLE
MGMT-9110: remove v1 UpdateCluster

### PR DESCRIPTION
**Pull request by slaviered**
## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory_test.go [hunk 1]
```diff
@@ -2579,46 +2578,16 @@ var _ = Describe("cluster", func() {
...
-				ClusterUpdateParams: &models.ClusterUpdateParams{
+				ClusterUpdateParams: &models.V2ClusterUpdateParams{
 					Name:          &newClusterName,
 					APIVipDNSName: swag.String("some dns name"),
 				},
 			})
 			Expect(reply).To(BeAssignableToTypeOf(common.NewApiError(http.StatusBadRequest, errors.Errorf("error"))))
 		})
 
-		It("cluster update failure on inventory refresh failure", func() {
```
Comment (nmagnezi) [inline internal/bminventory/inventory_test.go:2592]: Why was this test removed?

**Review by nmagnezi — CHANGES_REQUESTED** [review]
Looks good.
I just had one question about a test that got removed.
Also, you have a lint error to fix.

File: internal/bminventory/inventory_test.go (hunk 1 above)
Comment (slaviered) [reply internal/bminventory/inventory_test.go:2592]: a rebase error. fixed.

Comment (nmagnezi) [reply internal/bminventory/inventory_test.go:2592]: I still can't find this test in the recent version you uploaded

**Review by nmagnezi — CHANGES_REQUESTED** [review]
the test `"cluster update failure on inventory refresh failure"` is still missing

**Review by nmagnezi — APPROVED** [review]
/lgtm

**Review by nmagnezi — APPROVED** [review]
/lgtm

File: internal/bminventory/inventory_test.go (hunk 1 above)
Comment (slaviered) [reply internal/bminventory/inventory_test.go:2592]: In V2UpdateCluster there is currently no flow of refreshing hosts, so the entire test is not relevant for v2 (as for now)

**Pull request merged**